### PR TITLE
Rename service id to hash

### PR DESCRIPTION
**Pull request by NicolasMahe**
Depend on https://github.com/mesg-foundation/core/pull/627
Second half of https://github.com/mesg-foundation/core/issues/625

### CODE REVIEW

**Comment by NicolasMahe** [comment]
This PR introduce a one change on the api:
https://github.com/mesg-foundation/core/pull/626/files#diff-3ec4078a2b83ce9f8a52b41a8f1f4397R476

```
-  string ID = 10;                           // Service's unique id service hash.
+  string Hash = 10;                           // Service's unique id service hash.
```

@ilgooz @krhubert I wonder if it's a breaking change. The type, and the number are the same. Only the name change. What do you think?

**Comment by ilgooz** [comment]
> https://github.com/mesg-foundation/core/pull/626/files#diff-3ec4078a2b83ce9f8a52b41a8f1f4397R476

It shouldn't be a breaking change. Clients just will keep using _ID_ key. But for the clients that directly imports the pb file from _protobbuf_ pkg, it'll be a breaking change.

File: database/service_db.go [hunk 1]
```diff
@@ -13,14 +13,14 @@ import (
...
+	hashKeyPrefix = "hash_"
 )
 
 var (
-	errCannotSaveWithoutID  = errors.New("database: can't save service without id")
-	errCannotSaveWithoutSID = errors.New("database: can't save service without SID")
-	errSIDSameLen           = errors.New("database: service SID can't have same length as id")
+	errCannotSaveWithoutHash = errors.New("database: can't save service without hash")
+	errCannotSaveWithoutSID  = errors.New("database: can't save service without SID")
+	errSIDSameLen            = errors.New("database: service SID can't have same length as id")
```
Comment (ilgooz) [inline database/service_db.go]:
```suggestion
	errSIDSameLen            = errors.New("database: sid can't have the same length with hash")
```

File: protobuf/coreapi/api.proto [hunk 2]
```diff
@@ -473,7 +473,7 @@ message Service {
     RUNNING = 4;
   }
   
-  string ID = 10;                           // Service's unique id service hash.
+  string Hash = 10;                           // Service's unique id service hash.
```
Comment (ilgooz) [inline protobuf/coreapi/api.proto]:
```suggestion
  string Hash = 10;                           // Service's hash.
```

File: service/service.go [hunk 3]
```diff
@@ -24,8 +24,8 @@ import (
 
 // Service represents a MESG service.
 type Service struct {
-	// ID is the unique id of service.
-	ID string `hash:"-"`
+	// Hash is a generated string that uniquely identify a service.
```
Comment (ilgooz) [inline service/service.go]:
```suggestion
	// Hash is calculated from the combination of service's source and mesg.yml.
	// It represents the service uniquely.
```

File: database/service_db.go [hunk 4]
```diff
@@ -192,13 +192,13 @@ func (d *LevelDBServiceDB) Save(s *service.Service) error {
 	}
 
 	// save service with id
```
Comment (ilgooz) [inline database/service_db.go]:
```suggestion
	// save service with hash.
```

File: api/execute.go [hunk 5]
```diff
@@ -52,7 +52,7 @@ func (e *taskExecutor) checkServiceStatus(s *service.Service) error {
 		return err
 	}
 	if status != service.RUNNING {
-		return &NotRunningServiceError{ServiceID: s.ID}
+		return &NotRunningServiceError{ServiceID: s.Hash}
```
Comment (ilgooz) [inline api/execute.go]: here we use _ServiceID_ it should be _ServiceHash_ or _Hash_.

File: database/service_db.go [hunk 6]
```diff
@@ -13,14 +13,14 @@ import (
...
+	hashKeyPrefix = "hash_"
 )
 
 var (
-	errCannotSaveWithoutID  = errors.New("database: can't save service without id")
-	errCannotSaveWithoutSID = errors.New("database: can't save service without sid")
-	errSIDSameLen           = errors.New("database: sid can't have the same length as id")
+	errCannotSaveWithoutHash = errors.New("database: can't save service without hash")
+	errCannotSaveWithoutSID  = errors.New("database: can't save service without sid")
+	errSIDSameLen            = errors.New("database: sid can't have the same length as id")
```
Comment (ilgooz) [inline database/service_db.go]:
```suggestion
	errSIDSameLen            = errors.New("database: sid can't have the same length with hash")
```

File: database/service_db.go [hunk 7]
```diff
@@ -191,14 +191,14 @@ func (d *LevelDBServiceDB) Save(s *service.Service) error {
...
 
-	// save service with id
-	if err := tx.Put([]byte(idKeyPrefix+s.ID), b, nil); err != nil {
+	// save service with hash.
+	if err := tx.Put([]byte(hashKeyPrefix+s.Hash), b, nil); err != nil {
 		tx.Discard()
 		return err
 	}
 
 	// save sid-id pair of service.
```
Comment (ilgooz) [inline database/service_db.go]:
```suggestion
	// save sid-hash pair of service.
```

File: protobuf/coreapi/api.proto [hunk 8]
```diff
@@ -473,7 +473,7 @@ message Service {
     RUNNING = 4;
   }
   
-  string ID = 10;                           // Service's unique id service hash.
+  string hash = 10;                         // Service's unique id service hash.
```
Comment (ilgooz) [inline protobuf/coreapi/api.proto]:
```suggestion
  string hash = 10;                         // Service's hash.
```

File: api/execute.go (hunk 5 above)
Comment (NicolasMahe) [reply api/execute.go]: Let's leave it like this. From a user perspective, ServiceID is fine.
I'm also preparing another PR that remove completely Hash.

**Comment by ilgooz** [comment]
Need to re-gen proto

**Comment by NicolasMahe** [comment]
Close this PR in favor of https://github.com/mesg-foundation/core/pull/630

**Pull request closed without merging**